packages/iov-ethereum/src/ethereumconnection.spec.ts

```diff
@@ -300,12 +306,102 @@ describe("EthereumConnection", () => {
         .catch(err => expect(err).toMatch(/Header 99999999999999 doesn't exist yet/));
       connection.disconnect();
     });
   });
+
+  describe("feed", () => {
+    async function postTransaction(
+      quantity: string,
+      memo: string,
+      connection: EthereumConnection,
+    ): Promise<PostTxResponse> {
+      const wallet = Secp256k1HdWallet.fromMnemonic(
+        "oxygen fall sure lava energy veteran enroll frown question detail include maximum",
+      );
+      const mainIdentity = await wallet.createIdentity(HdPaths.bip44(60, 0, 0, 1));
+
+      const recipientAddress = "0xE137f5264b6B528244E1643a2D570b37660B7F14" as Address;
+
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        chainId: nodeChainId,
+        signer: mainIdentity.pubkey,
+        recipient: recipientAddress,
+        amount: {
+          quantity: quantity,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasPrice: {
+          quantity: gasPrice,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasLimit: {
+          quantity: gasLimit,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        memo: memo,
+      };
+      const senderAddress = ethereumCodec.keyToAddress(mainIdentity.pubkey);
+      const nonceResp = await connection.getNonce({ address: senderAddress });
+      const signingJob = ethereumCodec.bytesToSign(sendTx, nonceResp.data[0]);
+      const signature = await wallet.createTransactionSignature(
+        mainIdentity,
+        signingJob.bytes,
+        signingJob.prehashType,
+        nodeChainId,
+      );
+
+      const signedTransaction: SignedTransaction = {
+        transaction: sendTx,
+        primarySignature: {
+          nonce: nonceResp.data[0],
+          pubkey: mainIdentity.pubkey,
+          signature: signature,
+        },
+        otherSignatures: [],
+      };
+      const bytesToPost = ethereumCodec.bytesToPost(signedTransaction);
+
+      const resultPost = await connection.postTx(bytesToPost);
+      return resultPost;
+    }
+
+    describe("watchBlockHeaders", () => {
+      it("watches headers with same data as getBlockHeader", async () => {
+        pendingWithoutEthereum();
+        const connection = await EthereumConnection.establish(base, wsUrl);
+        const events = new Array<BlockHeader>();
+
+        const subscription = connection.watchBlockHeaders().subscribe({
+          next: info => {
+            events.push(info);
+            if (events.length === 2) {
+              expect(events[0].height).toEqual(events[1].height - 1);
+              connection.getBlockHeader(info.height).then(header => {
+                expect(header).toEqual(events[1]);
+                subscription.unsubscribe();
+                connection.unwatchBlockHeaders();
+                connection.disconnect();
```

Comment:
Good. Now move `connection.disconnect();` after the sleep. Otherwise the socket is disconnected directly after the second event and before the second `await postTransaction("5445500", "watch header test 2-" + new Date(), connection);` is done